src/main/java/ch/zhaw/psit4/services/implementation/SipClientServiceImpl.java

```diff
@@ -119,4 +129,26 @@ public void deleteSipClient(long id) {
             throw new SipClientDeletionException(message, e);
         }
     }
+
+    private Company getExistingCompany(SipClientDto sipClientDto) {
+        if (sipClientDto.getCompany() == null) {
+            String message = "Company was null";
+            LOGGER.error(message);
+            throw new CompanyRetrievalException(message);
+        }
+        if (sipClientDto.getCompany().getId() == null) {
+            String message = "Company id was null";
+            LOGGER.error(message);
+            throw new CompanyRetrievalException(message);
+        }
+        return companyRepository.findOne(sipClientDto.getCompany().getId());
+    }
+
+    private void checkCompanyExistence(SipClientDto sipClientDto, Company existingCompany) {
+        if (existingCompany == null) {
```

Comment:
This method has a side effect, i.e. it throws an exception. Make that clear by renaming the method, for instance `existsCompanyOrThrow()`. However, the database system should complain about non existing companies. So there is no need for doing it here as well.